File: learnsynth_offline_llm/android/src/main/kotlin/com/learnsynth/learnsynth_offline_llm/StubRunner.kt
package com.learnsynth.learnsynth_offline_llm

import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.Job
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class StubRunner : Runner {
    private var ready = false
    private var job: Job? = null
    private val scope = CoroutineScope(Dispatchers.Default)

    override suspend fun load(modelPath: String) {
        ready = true
    }

    override fun isReady(): Boolean = ready

    override fun generateStream(
        prompt: String,
        maxTokens: Int,
        temperature: Double,
        onToken: (String) -> Unit
    ) {
        job?.cancel()
        job = scope.launch {
            val text = "Android stub: $prompt"
            for (ch in text.toCharArray()) {
                onToken(ch.toString())
                delay(40)
            }
        }
    }

    override fun cancel() {
        job?.cancel()
    }
}

File: learnsynth_offline_llm/ios/Classes/StubRunner.swift
import Foundation

class StubRunner: Runner {
    private var ready = false
    private var task: Task<Void, Never>? = nil

    func load(modelPath: String) async {
        ready = true
    }

    func isReady() -> Bool { return ready }

    func generateStream(prompt: String, maxTokens: Int, temperature: Double, onToken: @escaping (String) -> Void) {
        task?.cancel()
        task = Task {
            let text = "iOS stub: \(prompt)"
            for ch in text {
                onToken(String(ch))
                try? await Task.sleep(nanoseconds: 40_000_000)
            }
        }
    }

    func cancel() {
        task?.cancel()
    }
}
